intel-sgx/dcap_artifact_retrieval/src/lib.rs

```diff
@@ -52,48 +42,21 @@ quick_error! {
             description("Intel certification services returned a header that could not be decoded")
             display("Failed to decode header")
         }
-        HeaderParseError(msg : &'static str) {
-            description("Header could not be parsed")
-            display("Failed to parse header {}", msg)
-        }
         CertificateParseError(msg: &'static str) {
             description("Certificate could not be parsed")
             display("Failed to parse certificate {}", msg)
         }
-        CertificateEncodingError(err: ASN1Error) {
-            from()
-        }
         NoEncPPID {
             description("Enc_ppid is required, but not provided")
             display("No enc_ppid was provided")
         }
-        NoCPUSVN {
-            description("CPU_svn is required, but not provided")
-            display("No cpu_svn was provided")
-        }
-        NoPCEISVSVN {
-            description("PCE ISVSVN is required, but not provided")
-            display("No pce_isvsvn was provided")
-        }
-        NoPCEID {
-            description("PCEID is required, but not provided")
-            display("No pce_id was provided")
-        }
         NoQeID {
             description("QEID is required, but not provided")
             display("No QE ID was provided")
         }
-        NoAPIKey {
-            description("PCS key is required, but not provided")
-            display("No api_key was provided")
-        }
-        OfflineAttestationError(err: OAError) {
+        PCSProvisioningError(err: pcs::Error) {
```

Comment:
@raoulstrackx This is the most generic name I came up with that I believe describes the logic of the 2 places this error variant is used in:
1) ```let tcbinfo = prov_client.tcbinfo(&fmspc)?;
        let tcbinfo_file = tcbinfo.store(output_dir).map_err(|e| Error::PCSProvisioningError(e))?;
```
2) fn parse_response(&self, response_body: String, response_headers: Vec<(String, String)>, _api_version: PcsVersion) -> Result<Self::Output, Error> {
        let ca_chain = parse_issuer_header(&response_headers, "SGX-PCK-Certificate-Issuer-Chain")?;
        PckCerts::parse(&response_body, ca_chain).map_err(|e| Error::PCSProvisioningError(e))
    }
```

Reply:
The first one is only used in the `main.rs`. So you can easily resolve that in another way.
The second is about errors in parsing the result you get from the PCS when fetching `PckCerts`. So `PckCertsParseError` or `BodyParseError` would make sense